File: frontend/src/core/domain/todo.ts

export interface Todo {
    id: number;
    user_id: number;
    title: string;
    description: string;
    status: 'pending' | 'in_progress' | 'done';
    image_path?: string;
    created_at: string;
    updated_at: string;
  }
  
  export interface CreateTodoRequest {
    title: string;
    description: string;
    status: string;
    image?: File;
  }
  
  export interface UpdateTodoRequest {
    title: string;
    description: string;
    status: string;
    image?: File;
  }

File: backend/internal/core/domain/todo.go
package domain

import "time"

type Todo struct {
	ID          int       `json:"id"`
	UserID      int       `json:"user_id"`
	Title       string    `json:"title"`
	Description string    `json:"description"`
	Status      string    `json:"status"`
	ImageID     string    `json:"image_id,omitempty"` // Changed from ImagePath to ImageID
	CreatedAt   time.Time `json:"created_at"`
	UpdatedAt   time.Time `json:"updated_at"`
}
type CreateTodoRequest struct {
	Title       string `json:"title"`
	Description string `json:"description"`
	Status      string `json:"status"`
}

type UpdateTodoRequest struct {
	Title       string `json:"title"`
	Description string `json:"description"`
	Status      string `json:"status"`
}